src/app/profiles.rs

```diff
@@ -127,7 +139,24 @@ where
             }
 
             match rx.poll() {
-                Ok(Async::NotReady) => return Async::NotReady,
+                Ok(Async::NotReady) => match hangup.poll() {
+                    Ok(Async::Ready(_)) => {
+                        // We should never get here, the hangup tx should never
+                        // be written to.
+                        warn!("profile stream hangup");
+                        return StreamState::SendLost.into();
```

Comment:
Consider making this `unreachable!()`, since it's technically a "programming error" if we enter this branch?

Reply:
Or you could make it literally impossible by making the message type `Never`. You can `match never {}`, the compiler is happy, and eliminates the branch entirely.

Reply:
I love types